Run processes in background

Jeff-Thompson12 opened the pull request:
Addresses #711

Jeff-Thompson12 committed:
Move report creation outside of download handler

Jeff-Thompson12 committed:
Move report creation to function

Jeff-Thompson12 committed:
Don't mandate progress in function

Jeff-Thompson12 committed:
Repair downloadHandler tests

Jeff-Thompson12 committed:
Repair reportPreview tests

Jeff-Thompson12 committed:
Run report as background process

Jeff-Thompson12 committed:
Fix some issues with the background process

Jeff-Thompson12 committed:
Only run in background if doing 4 or more packages

Jeff-Thompson12 committed:
Clean up some notes in R-CMD Check

Jeff-Thompson12 committed:
Merge branch 'dev' into jt-711-r_bg

Jeff-Thompson12 committed:
Add callr to imports

Jeff-Thompson12 committed:
Update progress details and timing

Jeff-Thompson12 committed:
Update progress from background R process

Jeff-Thompson12 committed:
Put details on separate line

Jeff-Thompson12 commented:
Looking for some feedback on this PR. Interested in what features you guys think should be added or what already needs to be changed.

Jeff-Thompson12 committed:
Resolve merge conflicts with `dev`

Merge branch 'dev' into jt-711-r_bg

# Conflicts:
#	R/mod_downloadHandler.R

Jeff-Thompson12 committed:
Update NEWS

Jeff-Thompson12 committed:
Resolve merge conflict with `dev`

Merge branch 'dev' into jt-711-r_bg

# Conflicts:
#	NEWS.md
#	R/mod_downloadHandler.R

Jeff-Thompson12 committed:
Merge branch 'dev' into jt-711-r_bg

Jeff-Thompson12 committed:
Update version number

AARON-CLARK reviewed (changes requested):
Hi @Jeff-Thompson12,

This is pretty sweet! In my first test, I ran html reports for 3 reports, then 5 reports. When downloading 3 reports, I noticed I was still able to do things in the app, which was not expected. Is that normal? 

![image](https://github.com/pharmaR/riskassessment/assets/17878953/d51cfa13-0a9a-45a2-9c26-1e53f0c71b83)

When rendering 5 reports, I was able to download a new package to the db while it was running, which is amazing. However, when it finished, the app crashed and there was an error in the console:

![image](https://github.com/pharmaR/riskassessment/assets/17878953/8091c796-02ed-4233-893a-5cd9a4965f1f)

jthompson-arcus committed:
Resolve merge conflict with `dev`

Merge branch 'dev' into jt-711-r_bg

# Conflicts:
#	DESCRIPTION
#	NEWS.md
#	R/mod_downloadHandler.R

jthompson-arcus committed:
Set a busy handler for `dbUpdate()` and `dbSelect()`

jthompson-arcus committed:
Remove notification when download is complete

jthompson-arcus committed:
Resolve merge conflicts with `dev`

Merge branch 'dev' into jt-711-r_bg

# Conflicts:
#	NEWS.md

jthompson-arcus committed:
Soft disable downloading reports if background process is going

jthompson-arcus committed:
Zip files in download handler

Jeff-Thompson12 commented:
@aclark02-arcus something else I forgot to mention is that since downloading a report is now a separate event from the report creation it is possible for the report downloader or another user to change outputs. This could be as low risk as adding comments but could also include actions like deleting a package from the database, which now that I mention it could crash the app.
